### PR TITLE
Port to jbuilder

### PR DESCRIPTION
**Pull request by mseri**
This splits the libraries into

- xen-api-client
- xen-api-client-lwt
- xen-api-client-async

and should allow an easier switch to `xapi-client` once the circular dependencies are resolved

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -2,18 +2,17 @@ language: c
 sudo: required
 install:
   - wget https://raw.githubusercontent.com/ocaml/ocaml-travisci-skeleton/master/.travis-opam.sh
-  - wget https://raw.githubusercontent.com/simonjbeaumont/ocaml-travis-coveralls/master/travis-coveralls.sh
 script:
   - bash -ex .travis-opam.sh
-  - if [ ! -z "${XS_COV}" ]; then bash -ex travis-coveralls.sh || true; fi
 env:
   global:
     - OCAML_VERSION=4.04
```
Comment (gaborigloi) [inline .travis.yml]: I guess we should update this to Docker in a future PR.

File: lib/jbuild [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+        (cmd ^ " | grep -oEe '-ppx (\"([^\"\\]|\\.)+\"|\\w+)'")
+    in
+    let rec go ic acc =
+      try go ic (acc ^ " " ^ input_line ic) with End_of_file -> close_in ic; acc
+    in
+    go ic ""
+
+
+let coverage_rewriter =
+  let is_coverage = try Unix.getenv "BISECT_ENABLE" = "YES" with Not_found -> false in
```
Comment (gaborigloi) [inline lib/jbuild]: Is this step unnecessary with the latest bisect? https://github.com/aantron/bisect_ppx/blob/master/doc/advanced.md#with-jbuilder

Comment (gaborigloi) [reply lib/jbuild]: Or is it to only include `preprocess` as necessary, for extra safety?

Comment (mseri) [reply lib/jbuild]: It is to avoid having to add `bisect_ppx` as a nonnecessary `opam` dependency

File: .travis.yml (hunk 1 above)
Comment (mseri) [reply .travis.yml]: Until `docker-opam` is fixed to properly support `BASE_REPO` I cannot update it

**Pull request merged**